Bottom: c3972894ac587422352563595bd7fbe982c7ae5e
Top:    7b6fb2c79abc1a0050f19a2aae0d0e1a9cc19220
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-12-24 01:40:34 +0000

djopt.c: Allocate correct size for structure pointer

(Caught by coverity)

Coverity-cid: 239


---

diff --git a/src/djopt.c b/src/djopt.c
index daeae0a..a377508 100644
--- a/src/djopt.c
+++ b/src/djopt.c
@@ -1331,7 +1331,7 @@ orthopull_1 (corner_s * c, int fdir, int rdir, int any_sel)
       if (ln >= lm)
 	{
 	  lm = ln + 10;
-	  ls = (line_s **) realloc (ls, lm * sizeof (line_s));
+	  ls = (line_s **) realloc (ls, lm * sizeof (line_s *));
 	}
       ls[ln++] = l;
       c2 = other_corner (l, c2);
